src/main/kotlin/com/openlattice/data/storage/PostgresDataQueries.kt

```diff
@@ -58,51 +64,43 @@ fun buildPreparableFiltersSql(
         metadataOptions: Set<MetadataOption>,
         linking: Boolean,
         idsPresent: Boolean,
-        partitionsPresent: Boolean
+        partitionsPresent: Boolean,
+        detailed: Boolean = false
 ): Pair<String, Set<SqlBinder>> {
     val filtersClauses = buildPreparableFiltersClause(startIndex, propertyTypes, propertyTypeFilters)
     val filtersClause = if (filtersClauses.first.isNotEmpty()) " AND ${filtersClauses.first} " else ""
-
     val metadataOptionColumns = metadataOptions.associateWith(::mapMetaDataToColumnSql)
-    val nonAggregatedMetadataSql = metadataOptionColumns
-            .filter { !isMetaDataAggregated(it.key) }.values.joinToString("")
-    val innerGroupBy = groupBy(ESID_EKID_PART_PTID + nonAggregatedMetadataSql)
-    // TODO: remove IS NOT NULL post-migration
-    val linkingClause = if (linking) {
-        " AND ${ORIGIN_ID.name} IS NOT NULL AND ${ORIGIN_ID.name} != '${IdConstants.EMPTY_ORIGIN_ID.id}' "
-    } else {
-        ""
-    }
+    val metadataOptionColumnsSql = metadataOptionColumns.values.joinToString("")
+    val innerGroupBy = groupBy(ESID_EKID_PART_PTID)
+    val outerGroupBy = if (metadataOptions.contains(MetadataOption.ORIGIN_IDS)) {
+        groupBy(ESID_EKID_PART)
+    } else groupBy("$ESID_EKID_PART,${ORIGIN_ID.name}")
+    val linkingClause = if (linking) " AND ${ORIGIN_ID.name} != '${ID_VALUE.name}' " else ""
```

Comment:
Maybe I'm missing here something, but right now, this will return true for both linking and non-linking entities + `upsertPropertyValueSql` did not change in a way, that it would insert the normal entity key id to the `origin_id` field.

Reply:
This needs a migration and I did forget to insert normal entity key is upsertEntities